crates/rattler_package_streaming/src/read.rs

```diff
@@ -76,13 +81,73 @@ pub fn extract_conda(reader: impl Read, destination: &Path) -> Result<ExtractRes
         // Take the file out of the [`ManuallyDrop`] to properly drop it.
         let _ = ManuallyDrop::into_inner(file);
     }
+    compute_hashes(md5_reader)
+}
+
+/// Extracts the contents of a `.conda` package archive by first fully dumping the reader
+/// into a in-memory buffer, instead of streaming the reader
+pub fn extract_conda_via_buffering(
+    reader: impl Read,
+    destination: &Path,
+) -> Result<ExtractResult, ExtractError> {
+    // delete destination first, as this method is usually used as a fallback from a failed streaming decompression
+    if destination.exists() {
+        std::fs::remove_dir_all(destination).map_err(ExtractError::CouldNotCreateDestination)?;
+    }
+    std::fs::create_dir_all(destination).map_err(ExtractError::CouldNotCreateDestination)?;
+
+    let mut buffer = Vec::new();
+    let sha256_reader = rattler_digest::HashingReader::<_, rattler_digest::Sha256>::new(reader);
+    let mut md5_reader =
+        rattler_digest::HashingReader::<_, rattler_digest::Md5>::new(sha256_reader);
+    md5_reader.read_to_end(&mut buffer)?;
 
+    // The stream must be seekable when decompressing zip archives using data descriptors
+    let cursor = Cursor::new(buffer);
+
+    let mut archive = ZipArchive::new(cursor)?;
```

Comment:
Not 100% sure of the tradeoff between having the package all in-memory vs first dumping it to a tempfile. Do you guys have an opinion on this?